bin/spice/cmd/login.go

```diff
@@ -192,26 +194,50 @@ spice login postgres engine --password <password>
 		passwordFlag: fmt.Sprintf("No password provided, use --%s or -p to provide a password", passwordFlag),
 	}, map[string]string{
 		passwordFlag: api.AUTH_PARAM_PG_PASSWORD,
-	}),
+	}, []string{}),
 }
 
 var snowflakeCmd = &cobra.Command{
 	Use:   "snowflake",
 	Short: "Login to a Snowflake warehouse",
 	Example: `
 spice login snowflake --account <account-identifier> --username <username> --password <password>
+spice login snowflake --account <account-identifier> --username <username> --private-key-path <private-key-path> --passphrase <passphrase>
 
 # See more at: https://docs.spiceai.org/
 `,
-	Run: CreateLoginRunFunc(api.AUTH_TYPE_SNOWFLAKE, map[string]string{
-		accountFlag:  fmt.Sprintf("No account identifier provided, use --%s or -a to provide an account identifier", accountFlag),
-		usernameFlag: fmt.Sprintf("No username provided, use --%s or -u to provide a username", usernameFlag),
-		passwordFlag: fmt.Sprintf("No password provided, use --%s or -p to provide a password", passwordFlag),
-	}, map[string]string{
-		accountFlag:  api.AUTH_PARAM_ACCOUNT,
-		usernameFlag: api.AUTH_PARAM_USERNAME,
-		passwordFlag: api.AUTH_PARAM_PASSWORD,
-	}),
+	Run: func(cmd *cobra.Command, args []string) {
+
+		if privateKeyPath, err := cmd.Flags().GetString(privateKeyPathFlag); err == nil && privateKeyPath != "" {
+
```

Comment:
TODO: do we need to resolve relative paths to absolute paths here to make sure paths are stored as absolute?